src/shapes/group.class.js

```diff
@@ -417,29 +417,6 @@
       this._drawClipPath(ctx, this.clipPath);
     },
 
-    /**
-     * Check if cache is dirty
-     */
-    isCacheDirty: function(skipCanvas) {
-      if (this.callSuper('isCacheDirty', skipCanvas)) {
-        return true;
-      }
-      if (!this.statefullCache) {
-        return false;
-      }
-      for (var i = 0; i < this._objects.length; i++) {
-        if (this._objects[i].isCacheDirty(true)) {
-          if (this._cacheCanvas) {
-            // if this group has not a cache canvas there is nothing to clean
-            var x = this.cacheWidth / this.zoomX, y = this.cacheHeight / this.zoomY;
-            this._cacheContext.clearRect(-x / 2, -y / 2, x, y);
-          }
-          return true;
-        }
-      }
-      return false;
-    },
```

Comment:
Why this goes all away?
https://github.com/fabricjs/fabric.js/pull/7920/files#diff-216444c1bc3837d560c8241c4d9643b6de8c147095081c5b38715b9e058c24cbL427
if is not statefulCache, return false, stop there.
The rest of the code was executed only if the group had `statefullCache` activated. Since it is no more possible to activate it, all of this function becomes just a callSuper, and so it can be removed.

Reply:
I don't understand and I don't even know what `statefulCache` is about

Reply:
statefulCache determine that there is a cache invalidation comparing all the properties of the object with a copy of those when the cache was created. It is slow when you have gradients or large groups or a long path.

Reply:
I'm not sure I need to answer this question.
But group invalidation is done in `Object#_set`, this piece of code is an artifact, and a damaging one too.

Reply:
it wasn't a question :D it was more what i was trying to answer to with the comment.
I would expect a reader does not get automatically why a large chunk of code is deleted

Reply:
OK now I get it
Question:
> Why this goes all away?

Answer:
> https://github.com/fabricjs/fabric.js/pull/7920/files#diff-216444c1bc3837d560c8241c4d9643b6de8c147095081c5b38715b9e058c24cbL427